lib/experimental/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -108,8 +108,12 @@ public static function get_block_data() {
 		$blocks   = $registry->get_all_registered();
 		$config   = array( 'version' => 1 );
 		foreach ( $blocks as $block_name => $block_type ) {
-			if ( isset( $block_type->supports['__experimentalStyle'] ) ) {
-				$config['styles']['blocks'][ $block_name ] = static::remove_JSON_comments( $block_type->supports['__experimentalStyle'] );
+			if ( is_array( $block_type->style ) ) {
```

Comment:
This change is in the experimental folder, but this logic is required for the Button block to work correctly, so we need to include it in `compat/wordpress-6.0` folder.